### PR TITLE
add service accounts to registry and enable registry-based authorizer

### LINKED ISSUE #1976: Add identity registry support for service accounts registered via `istioctl register`

### PR DESCRIPTION
**Pull request by crazytan**
**What this PR does / why we need it**:
This PR follows https://github.com/istio/istio/pull/1988 and let istio CA gRPC server uses registry-backed authorizor by default. The purpose is to enable node agent running on GCE instances to use native JWT token to get key/cert pair for workload. To achieve this I made three additional changes:

1. add a service account controller for identity registry. The controller adds the SpiffeIDs of all service accounts to the registry. This ensures that Istio mesh expansion still works (https://istio.io/docs/setup/kubernetes/mesh-expansion.html).
2. change the `idTokenAuthenticator` and use "email" field in JWT to populate the identity, which contains the actual GCP service account name.
3. if authorization on JWT token passes, add its requested IDs to the registry. This is to ensure that the subsequent certificate rotation still works.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #1976

**Special notes for your reviewer**: I'm not sure whether to write release note...

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: security/cmd/istio_ca/main.go [hunk 1]
```diff
@@ -142,13 +143,24 @@ func runCA() {
...
-	serviceController := controller.NewServiceController(cs.CoreV1(), opts.namespace,
-		registry.K8SServiceAdded, registry.K8SServiceDeleted, registry.K8SServiceUpdated)
-	serviceController.Run(stopCh)
-
 	if opts.grpcPort > 0 {
+		// start registry if gRPC server is to be started
+		reg := registry.GetIdentityRegistry()
+		ch := make(chan struct{})
+
+		// monitor service objects for `istioctl register` command
```
Comment (wattli) [inline security/cmd/istio_ca/main.go]: Are we actually monitor the istioctl register command or monitor all services in K8s?

File: security/pkg/registry/kube/service.go [hunk 2]
```diff
@@ -23,32 +24,30 @@ import (
...
+
+	"istio.io/istio/pilot/platform/kube"
+	"istio.io/istio/security/pkg/registry"
 )
 
-// ServiceController monitors the service definition changes in a namespace.
-// Callback functions are called whenever there is a new service, a service deleted,
-// or a service updated.
+// ServiceController monitors the service definition changes in a namespace. If a
+// new service is added with "alpha.istio.io/kubernetes-serviceaccounts" annotation
```
Comment (wattli) [inline security/pkg/registry/kube/service.go]: Note there is a legacy issue in my previous PR, ideally we want to use CanonicalServiceaccount... as mentioned here for non-k8s service account. https://github.com/istio/istio/blob/a1ba1fbf9cdeac6bd51898732f787948dcc9d160/pilot/platform/kube/conversion.go#L43

But as of now, let's just make it work and adjust later.

File: security/pkg/registry/kube/service.go [hunk 3]
```diff
@@ -76,13 +75,26 @@ func (c *ServiceController) Run(stopCh chan struct{}) {
 }
 
 func (c *ServiceController) serviceAdded(obj interface{}) {
-	c.addFunc(obj.(*v1.Service))
+	svc := obj.(*v1.Service)
```
Comment (wattli) [inline security/pkg/registry/kube/service.go]: You might also want to add the RBAC rules for K8s services at somewhere here https://github.com/istio/istio/blob/1f601f11ad63e27c730620b83a1b1506c5452c08/install/kubernetes/templates/istio-rbac-beta.yaml.tmpl#L69

File: security/cmd/istio_ca/main.go (hunk 1 above)
Comment (crazytan) [reply security/cmd/istio_ca/main.go]: service objects, in the same namespace as CA monitors. I'll change the wording.

File: security/pkg/registry/kube/service.go (hunk 2 above)
Comment (crazytan) [reply security/pkg/registry/kube/service.go]: Cool. I'll add a TODO

File: security/pkg/registry/kube/service.go (hunk 3 above)
Comment (crazytan) [reply security/pkg/registry/kube/service.go]: Will do. thanks for reminding

**Pull request merged**